packages/rfw/README.md

```diff
@@ -208,7 +328,409 @@ class _ExampleState extends State<Example> {
+events. For example, here is how the "7" button from the
+[calculator example](https://github.com/flutter/packages/blob/main/packages/rfw/example/wasm/logic/calculator.rfwtxt)
+is represented:
+
+```
+CalculatorButton(label: "7", onPressed: event "digit" { arguments: [7] }),
+```
+
+This creates a `CalculatorButton` widget with two arguments, `label`,
+a string, and `onPressed`, an event, whose name is "digit" and whose
+arguments are a map with one key, "arguments", whose value is a list
+with one value 7.
+
+When the button is pressed, an event with the name "digit" and the
+given arguments is reported to the `RemoteWidget`'s `onEvent`
+callback. That callback takes two arguments, the event name and the
+evnet arguments.
+
+On the implementation side, in local widget libraries, arguments like
+the "onPressed" argument above must be turned into a Dart callback.
+
+The simplest kind of callback is a `VoidCallback` (no arguments, no
+return value). To turn an `event` value in a widget's arguments in the
+remote widget library into a `VoidCallback` in Dart that reports the
+event as described above, the `DataSource.voidHandler` method can be
+used. For example, here is a simplified `GestureDetector` that just
+implements `onTap`:
+
+```
+    return GestureDetector(
+      onTap: source.voidHandler(['onTap']),
+      child: source.optionalChild(['child']),
+    );
+```
+
+Sometimes, a callback has a different signature, in particular, it may
+provide arguments. To convert the `event` value into a Dart callback
+that reports an event as described above, the `DataSource.handler`
+method is used.
+
+In addition to the list of keys that identify the `event` value, the
+method takes a callback. The callback's purpose is to convert the
+given `trigger` (which, when called, reports the event) into the kind
+of callback the real `Widget` expects. This is usually written
+something like the following:
+
+```dart
+    return GestureDetector(
+      onTapDown: source.handler(['onTapDown'], (HandlerTrigger trigger) => (TapDownDetails details) => trigger()),
+      child: source.optionalChild(['child']),
+    );
+```
+
+To break this down more clearly:
+
+```dart
+    return GestureDetector(
+      // onTapDown expects a function that takes a TapDownDetails
+      onTapDown: source.handler<GestureTapDownCallback>( // this returns a function that takes a TapDownDetails
+        ['onTapDown'],
+        (HandlerTrigger trigger) { // "trigger" is the function that will send the event to RemoteWidget.onEvent
+          return (TapDownDetails details) { // this is the function that is returned by handler() above
+            trigger(); // the function calls "trigger"
+          };
+        },
+      ),
+      child: source.optionalChild(['child']),
+    );
+```
+
+In some cases, the arguments sent to the callback (the
+`TapDownDetails` in this case) are useful and should be passed to the
+`onEvent` as part of its arguments. This can be done by passing some
```

Comment:
```suggestion
`RemoteWidget.onEvent` as part of its arguments. This can be done by passing some
```

helps make clear whose "onEvent" we're talking about, and e.g. that this isn't meant as a metavariable for "onTap" / "onPressed" / etc.

Reply:
fixed